x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/edit_connector_flyout/gen_ai/use_get_dashboard.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useEffect, useRef, useCallback } from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { useKibana } from '../../../../..';
+import { getDashboard, getDashboardId } from './helpers';
+
+interface Props {
+  connectorId: string;
+}
+
+export interface UseGetDashboard {
+  dashboardUrl: string | null;
+  isLoading: boolean;
+}
+
+export const useGetDashboard = ({ connectorId }: Props): UseGetDashboard => {
```

Comment:
this is implemented in `stack_connectors`, but we cannot import from that plugin here. so i duplicated the code